crates/rust-client/src/store/data_store.rs

```diff
@@ -134,23 +142,22 @@ impl DataStore for ClientDataStore {
         map_root: Word,
         map_key: Word,
     ) -> Result<miden_objects::account::StorageMapWitness, DataStoreError> {
-        // TODO: Refactor the store call to be able to retrieve by map root.
-        let account_storage = self.store.get_account_storage(account_id).await?;
+        let account_storage = self
+            .store
+            .get_account_storage(account_id, AccountStorageFilter::Root(map_root))
+            .await?;
 
-        for slot in account_storage.slots() {
-            if let StorageSlot::Map(map) = slot
-                && map.root() == map_root
-            {
+        match account_storage.slots().first() {
+            Some(StorageSlot::Map(map)) => {
                 let witness = map.open(&map_key);
-                return Ok(witness);
-            }
+                Ok(witness)
+            },
+            _ => Err(DataStoreError::Other {
+                error_msg: format!("did not find map with {map_root} as a root for {account_id}")
+                    .into(),
+                source: None,
+            }),
```

Comment:
What do you think of this? To provide a bit more context
```suggestion
            slot => {
                let found_slot = slot
                    .map(|slot| {
                        let value_hex = slot.value().to_hex();
                        format!("Found StorageSlot::Value with {value_hex} as its value.")
                    })
                    .unwrap_or_else(|| format!("Storage Slot was empty"));

                Err(DataStoreError::Other {
                    error_msg: format!(
                        "Did not find map with {map_root} as a root for {account_id}: {found_slot}"
                    )
                    .into(),
                    source: None,
                })

```